workflow-manager/src/domain/cosmos_cw.rs

```diff
@@ -325,18 +329,23 @@ impl Connector for CosmosCosmwasmConnector {
         .build_any();
 
         self.wallet
-            .simulate_tx(vec![m])
-            // .broadcast_tx(vec![m], None, None, BroadcastMode::Sync) // TODO: change once we ready
+            // .simulate_tx(vec![m])
+            .broadcast_tx(vec![m], None, None, BroadcastMode::Sync)
             .await
-            .map(|_| ())
-            .context("Failed to broadcast the TX")
-            .map_err(|e| CosmosCosmwasmError::Error(e).into())
+            .map_err(CosmosCosmwasmError::Error)?;
+
+        sleep(std::time::Duration::from_secs(12)).await;
```

Comment:
question regarding this, can't we query the TX hash?
this is how local-IC does it for example
```rust
        for _ in 0..5 {
            let data = self.query_tx_hash(&tx_hash);

            if !data.to_string().starts_with("{\"error\":") {
                return Ok(data);
            }

            std::thread::sleep(std::time::Duration::from_secs(1));
        }

Reply:
Yes we can, I wanted to make sure i unblock the workflow, and const waiting time is the fastest for now.

Reply:
Opened an issue to not forget #50

Reply:
@keyleu @Art3miX Yeah the sleep is not great, nor the looping + sleep tbh. I think the correct pattern would be to use a wait handle to block the current thread until either, the transaction is finalized or errors out (need to subscribe to chain events before broadcasting the tx .. I'm not sure what's the correct API for that), or there is a timeout of x seconds. Basically, you want to wait asynchronously until some events occur, not wait synchronously and poll for result.